pkg/config/genesis_extensions.go

```diff
@@ -1,32 +1,61 @@
 package config
 
 import (
+	"encoding/base64"
 	"fmt"
 
 	"github.com/nspcc-dev/neo-go/pkg/core/native/noderoles"
 	"github.com/nspcc-dev/neo-go/pkg/crypto/keys"
+	"github.com/nspcc-dev/neo-go/pkg/util"
 )
 
 // GenesisExtensions represents the extensions enabled in the genesis block or by the
 // moment of native contracts initialization.
 type GenesisExtensions struct {
 	// Roles contains the set of roles that should be designated on native
 	// Designation contract initialization.
-	Roles map[noderoles.Role]keys.PublicKeys
+	Roles        map[noderoles.Role]keys.PublicKeys
+	Transactions []GenesisTransaction
 }
 
-// genesisExtensionsAux is an auxiliary structure for GenesisExtensions YAML marshalling.
-type genesisExtensionsAux struct {
-	Roles map[string]keys.PublicKeys `yaml:"Roles"`
+// GenesisTransaction is a placeholder for script that should be included into genesis
+// block as a transaction script with the given signers and system fee. Provided
+// system fee value will be taken from the stand by validators account which is
+// added to the list of Signers as a sender with None scope.
+type GenesisTransaction struct {
+	Script    []byte
+	Signers   []util.Uint160 // TODO: if needed, replace with []transaction.Signer and write YAML marshaller for it.
+	SystemFee int64
 }
 
+type (
+	// genesisExtensionsAux is an auxiliary structure for GenesisExtensions YAML marshalling.
+	genesisExtensionsAux struct {
+		Roles        map[string]keys.PublicKeys `yaml:"Roles"`
+		Transactions []genesisTransactionAux    `yaml:"Transactions"`
+	}
+	genesisTransactionAux struct {
+		Script    string         `yaml:"Script"`
+		Signers   []util.Uint160 `yaml:"Signers"`
```

Comment:
Can be simplified by using validator multisig-signed transactions only, no one else has any GAS anyway.

Reply:
Isn't it possible that some scoped witness of other account will be required during deploy? Not for the transfer, but for some other action.

Reply:
In fact, with this mechanism we don't even need Roles extension, because roles designation may be done via a separate genesis transaction with Committee signer set in this configuration. So GenesisTransactions is more generic mechanism.

Reply:
We can imagine a lot of things, but practically it's genesis, we have validators, we have committee, we don't have much else. And we better not have tens of transactions doing lots of things, otherwise it'd be a hell to configure.

Reply:
Fixed.